src/components/FormType.tsx

```diff
@@ -0,0 +1,126 @@
+import { useContext } from "react"
+import { ModalContext } from "~/state/ModalContext"
+
+type ModalStateSetter = (value: { open: boolean }) => void
+
+interface GitHubFormTypeProps {
+  modals: {
+    setCreateModalState: ModalStateSetter
+    setRulesetModalState: ModalStateSetter
+  }
+}
+
+const FormType = () => {
+  
+  const { eventTriggerState, setRulesetModalState, setCreateModalState } = useContext(ModalContext);
+
+  return (
+    <div>
+      {eventTriggerState === "mail" && <MailFormType />}
```

Comment:
for example maybe here

Reply:
yeah that flew over my head. will get that done 👍🏻